xapian-core/weight/tfidfweight.cc

```diff
@@ -136,7 +142,8 @@ TfIdfWeight::get_maxpart() const
 {
     Xapian::termcount wdf_max = get_wdf_upper_bound();
     Xapian::termcount len_min = get_doclength_lower_bound();
-    double wdfn = get_wdfn(wdf_max, len_min, normalizations[0]);
+    Xapian::termcount len_max = get_doclength_upper_bound();
+    double wdfn = get_wdfn(wdf_max, len_min, len_max, normalizations[0]);
     return get_wtn(wdfn * idfn, normalizations[2]) * wqf_factor;
```

Comment:
@ojwb I'd appreciate if you could clarify a bit more based on my previous comment? Thanks!